File: showcases/flair.ts
// Browser showcase for the animated Flair FBX asset.
//
// The existing flair.ts example loads the converted glTF asset.  This module
// keeps the original FBX path visible as a separate browser demonstration.
export async function flair(viewer: any) {
  viewer.setBackgroundColor([1, 1, 1])
  // Encode the space in the public filename for static-server compatibility.
  const animations = await viewer.loadFbx('/models/Flair.fbx')

  // Keep the camera fixed while the imported skeleton drives the mesh.
  viewer.resetCamera()

  // The first imported FBX clip is exposed as animation zero in JavaScript.
  if (animations.length > 0) {
    viewer.selectAnimation(0)
    await viewer.setMotionLines({ algorithm: 'uniform', count: 128, fps: 30 })
    viewer.setAnimationSpeed(1.0)
    viewer.playAnimation()
  }
}

File: scripts/previews/flair.lua
set_background_color({ 1, 1, 1 })
local animations = load_fbx("../public/models/Flair.fbx")
reset_camera()
if #animations > 0 then
  select_animation(1)
  set_animation_time(0.4)
  set_motion_lines_uniform(128, 30)
end
show_scene()
save_screenshot("../public/previews/flair.png")
